base/errorshow.jl

```diff
@@ -1086,6 +1092,36 @@ end
 
 Experimental.register_error_hint(methods_on_iterable, MethodError)
 
+# Display a hint in case the use tries to access non-member fields of container type datastructures
+function member_access_handler(io, exc)
+    @nospecialize
+    x = exc.x
+    objType = exc.objType
+    if objType <: Dict
+        println(io,
+            """
+            The field `$x` is not a member of Dict type. In case you
+            are trying to access values using keys consider
+            using `indexing` operation.
+            Example:
+                ```julia
+                dict = Dict($(x)=>1)
+                dict[$(x)]
+                ```
+            """
+        )
+    else
+        println(io,
+            """
+            This datatype doesn't allow member access of this field-`$(x)` for end users.
+            Please refer to documentation on $(objType) on access rights and means to do it.
+            """
+        )
```

Comment:
```suggestion
```

The Dict hint is worth adding; however, the non-dict case doesn't benefit from this hint IMO. Also, fields are private by default (Julia-private, not Java-private), so the lack of a getfield error does not mean that field access is allowed for end-users.

Reply:
Agree